lib/pharos/phase.rb

```diff
@@ -24,16 +35,18 @@ def self.register_component(component)
 
     # @param host [Pharos::Configuration::Host]
     # @param config [Pharos::Config]
-    # @param ssh [Pharos::SSH::Client]
     # @param master [Pharos::Configuration::Host]
-    def initialize(host, config: nil, ssh: nil, master: nil, cluster_context: nil)
+    def initialize(host, config: nil, master: nil, cluster_context: nil)
       @host = host
       @config = config
-      @ssh = ssh
       @master = master
       @cluster_context = cluster_context
     end
 
+    def ssh
+      @ssh ||= host.ssh
```

Comment:
Host already caches `ssh`, no need to do it here?

Reply:
No need really, but gets a direct reference instead of having to do a method lookup + invocation on 
`host` every time.

Reply:
I would then remove it from here, I'm quite sure that we don't really see any difference in performance.